README.md

```diff
@@ -123,22 +123,40 @@ ln -s /usr/bin/vim /usr/bin/vi
 		* `cd ../..`
 	1. Sanity checking for consistency in series file
 		* `./devutils/check_patch_files.sh`
-1. Check for esbuild dependency changes in file `build/src/DEPS` and adapt `downloads.ini` accordingly
-1. Check for commit hash changes of `src` submodule in `third_party/microsoft_dxheaders` (e.g. using GitHub https://github.com/chromium/chromium/tree/127.0.6533.72/third_party/microsoft_dxheaders) and adapt `downloads.ini` accordingly
-1. Check for version changes of windows rust crate (`third_party/rust/windows_x86_64_msvc/`) and adapt `downloads.ini` and `patches/ungoogled-chromium/windows\windows-fix-building-with-rust.patch` accordingly
-1. Use git to add changes and commit
+1. Use Git to add changes and commit
+
+### Update dependencies
+
+**NOTE:** For all steps, update `downloads.ini` accordingly.
+
+1. Check the [LLVM GitHub](https://github.com/llvm/llvm-project/releases/) for the latest version of LLVM.
+1. Check the esbuild version in file `build/src/third_party/devtools-frontend/src/DEPS` and find the closest release in the [esbuild GitHub](https://github.com/evanw/esbuild/releases) to it.
+	* Example: `version:3@0.24.0.chromium.2` should be `0.24.0`
+1. Check the [ninja GitHub](https://github.com/ninja-build/ninja/releases/) for the latest version of ninja.
+	1. Download the `ninja-win.zip` file.
+	1. Get the SHA-512 checksum using `sha512sum` in **`MSYS2 MSYS`**.
+1. Check the [Git GitHub](https://github.com/git-for-windows/git/releases/) for the latest version of Git.
+	1. Get the SHA-256 checksum for `PortableGit-<version>-64-bit.7z.exe`.
+1. Check for commit hash changes of `src` submodule in `third_party/microsoft_dxheaders` (e.g. using GitHub `https://github.com/chromium/chromium/tree/<version>/third_party/microsoft_dxheaders`).
+	1. Replace `version` with the Chromium version in `ungoogled-chromium/chromium_version.txt`.
+1. Check the [NodeJS website](https://nodejs.org/en/download) for the latest **LTS** version of NodeJS.
+	1. Download the "Standalone Binary" version.
+	1. Get the SHA-256 checksum using `sha256sum` in **`MSYS2 MSYS`**.
+1. Check for version changes of windows rust crate (`third_party/rust/windows_x86_64_msvc/`).
+	1. Update `patches/ungoogled-chromium/windows/windows-fix-building-with-rust.patch` accordingly.
```

Comment:
I am not sure if we should update those dependencies without having the need to do so.

Reply:
Keeping dependencies up-to-date is usually good practice, even if it seems as though we don't need to for some of them. We know LLVM and Rust have to be updated often, since Google often uses nightly builds of both.

Of the dependencies, Git and Ninja need to be updated the least often. Git could be removed entirely, given that we require it to be installed on the system anyway and we should not have to download a portable version of it.

Reply:
I agree that keeping rust and clang up to date is a good idea in genral. Rust is updated on each major release so we are good there but we had some issues with updating clang in the past, therefore I tend to keep a working version as long as possible.

The same is true for the other downloads that do not have hard coded dependencies in the chromium sources. Updating them as needed might save some headaches which might come with more recent versions.

Reply:
For the most part, we should use whichever version of a dependency Google is using, though that's often the latest version. Presumably, that's the version they either recommend or are currently having the fewest problems with, though it's not explicitly required.